### ISSUE TITLE
[cy.origin] Searching in`https://everfihelp.force.com/foundry/s/` in a cy.origin block hangs the browser.

### ISSUE THREAD

**Issue by mjhenkes** [cypress-io/cypress#21572]
### Current behavior

The following test hangs the browser when run in a cy.origin block

```javascript
it('visits force.com in a cy.origin block', () => {
    cy.visit('https://platform.everfi.net/')
    cy.origin('https://force.com/', () => {
      cy.visit('https://everfihelp.force.com/foundry/s/')
      cy.get('.search-input.search-input--right').type('blah')
      cy.get(
        '.slds-button.slds-button_icon.search-inputSearchButton.search-inputSearchButton--right.slds-button_icon-border',
      ).click()

      // cy.wait(2000);
      cy.contains('No results for "blah" in Knowledge').should('exist')
    })
  })
```

But when run by itself, the test executes correctly.

```javascript
  it('visits force.com directly', () => {
    cy.visit('https://everfihelp.force.com/foundry/s/')
    cy.get('.search-input.search-input--right').type('blah')
    cy.get(
      '.slds-button.slds-button_icon.search-inputSearchButton.search-inputSearchButton--right.slds-button_icon-border',
    ).click()

    // cy.wait(2000);
    cy.contains('No results for "blah" in Knowledge').should('exist')
  })
```

I've narrowed the issue down to have something to do with actually clicking the search button and the test will run correctly if that is commented out.

```javascript
  it('visits force.com in a cy.origin block', () => {
    cy.visit('https://platform.everfi.net/')
    cy.origin('https://force.com/', () => {
      cy.visit('https://everfihelp.force.com/foundry/s/')
      cy.get('.search-input.search-input--right').type('blah')
      // cy.get(
      //   '.slds-button.slds-button_icon.search-inputSearchButton.search-inputSearchButton--right.slds-button_icon-border',
      // ).click()

      // // cy.wait(2000);
      // cy.contains('No results for "blah" in Knowledge').should('exist')
    })
  })
```

### Desired behavior

Cypress should not hang when testing.

### Test code to reproduce

See above.

### Cypress Version

9.6.1

**Comment by AtofStryker** [comment]
Once #20685 is resolved we should revisit this to see if the issue is resolved

**Comment by idanElitzur** [comment]
Still happens to me with cypress 10.7.0.
Can you please post some reference how to handle this issue?
thanks

**Comment by AtofStryker** [comment]
Hey @idanElitzur. We currently do not have a workaround for this issue. We have spent a few weeks investigating this issue, but haven't been able to determine a root cause. Definitely a frustrating one. We have some significant improvements coming to `cy.origin` in the coming weeks, and what I am hoping is that some of the changes either help resolve this issue, or eliminate a few things that we know are problems. I have tried this with #23297 and didn't see any improvements, but as we gain further developments on this we will update the issue!
